### PR TITLE
Picker update items when children is changed

### PR DESCRIPTION
**Pull request by adids1221**
## Description
Picker supports updating the items list.

## Changelog
Picker supports updating the items list.

### CODE REVIEW

**Comment by adids1221** [comment]
@M-i-k-e-l 
Slack thread: https://wix.slack.com/archives/C3B1EMVF0/p1683729440903229

Snippet code for the playground screen if you want to test it:
`const options1 = [
  {
    id: '9a5d83fd-8570-482e-81ab-cfa88942ee60',
    name: 'Restaurant items (Default)'
  },
  {
    id: '16292b1f-7498-475d-a7bc-9204d9508fb4',
    name: 'vvvvv',
    revision: '1',
    createdDate: '2023-05-11T12:13:43.498Z',
    updatedDate: '2023-05-11T12:13:43.498Z'
  }
];

const options2 = [
  {
    id: '9a5d83fd-8570-482e-81ab-cfa88942ee60B',
    name: 'Restaurant Dining Stuff'
  },
  {
    id: '16292b1f-7498-475d-a7bc-9204d9508fb4A',
    name: 'BBBBBBB',
    revision: '1',
    createdDate: '2023-05-11T12:13:43.498Z',
    updatedDate: '2023-05-11T12:13:43.498Z'
  }
];

export default class PlaygroundScreen extends Component {
  state = {
    selectedValue: options1[0].name,
    items: options1
  };

  onChange = val => this.setState({selectedValue: val, items: [...options1, ...options2]});

  render() {
    const {selectedValue, items} = this.state;
    return (
      <View bg-grey80 flex padding-20>
        <View marginT-20>
          <Picker
            value={selectedValue}
            enableModalBlur={false}
            onChange={this.onChange}
          >
            {Object.values(items).map(taxGroup => (
              <Picker.Item
                key={taxGroup.id}
                value={taxGroup.id!} // TODO: Understand the cast
                label={taxGroup.name!} // TODO: Understand why '!' is needed
              />
            ))}
          </Picker>
        </View>
      </View>
    );
  }
}
`

**Review by M-i-k-e-l — COMMENTED** [review]
Sorry for taking this long...

**Comment by ixartz** [comment]
Thank you for maintaining this project, really love the library.

I second this, I also get the same issue. I have a dynamic list of items and they not updated when the list change.
Couldn't find any workaround.

Thank you in advance.

File: src/components/picker/types.tsx [hunk 1]
```diff
@@ -180,6 +180,10 @@ export type PickerBaseProps = Omit<NewTextFieldProps, 'value' | 'onChange'> &
      */
     testID?: string;
     children?: ReactNode | undefined;
+    /**
+     * Data source for Picker
+     */
+    items?: Pick<PickerItemProps, 'label' | 'value'>[];
```
Comment (M-i-k-e-l) [inline src/components/picker/types.tsx:186]: Personally I prefer the `testID` \ `children` to be last, your choice

File: src/components/picker/PickerItemsList.tsx [hunk 2]
```diff
@@ -66,7 +67,15 @@ const PickerItemsList = (props: PickerItemsListProps) => {
   },
   [children]);
 
+  const renderPropItems = useCallback(({item}: ListRenderItemInfo<PickerItemProps>) => {
+    return <PickerItem key={item.value} value={item.value} label={item.label} disabled={item.disabled}/>;
+  },
+  [items]);
+
   const renderList = () => {
+    if (items) {
```
Comment (M-i-k-e-l) [inline src/components/picker/PickerItemsList.tsx:76]: I don't really like adding another return, but I can't say I can think of a better alternative.

**Review by M-i-k-e-l — COMMENTED** [review]
Nice work, added a few comments

File: src/components/picker/PickerItemsList.tsx [hunk 3]
```diff
@@ -66,7 +67,15 @@ const PickerItemsList = (props: PickerItemsListProps) => {
   },
   [children]);
 
+  const renderPropItems = useCallback(({item}: ListRenderItemInfo<PickerItemProps>) => {
+    return <PickerItem key={item.value} value={item.value} label={item.label} disabled={item.disabled}/>;
+  },
+  [items]);
```
[the remark this answers is not in the document]
Comment (adids1221) [reply src/components/picker/PickerItemsList.tsx:73]: You are right, I'll remove the dependency.

File: src/components/picker/api/picker.api.json [hunk 4]
```diff
@@ -92,7 +92,8 @@
       "description": "Pass props to the list component that wraps the picker options (allows to control FlatList behavior)"
     },
     {"name": "pickerModalProps", "type": "ModalProps", "description": "Pass props to the picker modal"},
-    {"name": "useSafeArea", "type": "boolean", "description": "Add safe area in the Picker modal view"}
+    {"name": "useSafeArea", "type": "boolean", "description": "Add safe area in the Picker modal view"},
+    {"name": "items", "type": "PickerItemProps[]", "description": "Data source for Picker"}
```
[the remark this answers is not in the document]
Comment (adids1221) [reply src/components/picker/api/picker.api.json:96]: does this looks right: `"type": "{label: string, value: PickerSingleValue}[]"`

File: src/components/picker/PickerItemsList.tsx [hunk 5]
```diff
@@ -66,7 +67,14 @@ const PickerItemsList = (props: PickerItemsListProps) => {
   },
   [children]);
 
+  const renderPropItems = useCallback(({item}: ListRenderItemInfo<PickerItemProps>) => {
+    return <PickerItem value={item.value} label={item.label} disabled={item.disabled}/>;
```
Comment (M-i-k-e-l) [inline src/components/picker/PickerItemsList.tsx:71]: This can be simpler:
```suggestion
    return <PickerItem {...item}/>;
```

Comment (M-i-k-e-l) [reply src/components/picker/PickerItemsList.tsx:71]: And also, you're supporting `disabled` here but not in the type (and API)

Comment (M-i-k-e-l) [reply src/components/picker/PickerItemsList.tsx]: I think you missed this one

**Review by M-i-k-e-l — APPROVED** [review]

**Pull request merged**